lib/src/shared/main/java/com/couchbase/lite/Blob.java

```diff
@@ -376,224 +411,169 @@ public String digest() {
-                                throw new IllegalStateException(
-                                    "No data available to write for install.  Please ensure that all blobs in the "
-                                        + "document have non-null content.");
-                            }
-                            while ((bytesRead = contentStream.read(buffer)) > 0) {
-                                this.blobLength += bytesRead;
-                                blobOut.write(Arrays.copyOfRange(buffer, 0, bytesRead));
-                            }
-                        }
-                        finally {
-                            try {
-                                if (contentStream != null) { contentStream.close(); }
-                            }
-                            catch (IOException ignore) { }
-                        }
-                        key = blobOut.computeBlobKey();
-                        blobOut.install();
-                    }
-                    finally {
-                        blobOut.close();
-                    }
-                }
-                this.blobDigest = key.toString();
-                this.database = db;
-            }
-            finally {
-                if (store != null) { store.free(); }
-            }
+            store = db.getBlobStore();
+
+            key = (content != null) ? store.create(content) : storeBytesInDatabase(store);
+
+            this.blobDigest = key.toString();
+            this.database = db;
         }
         catch (Exception e) {
             throw new IllegalStateException(e);
         }
         finally {
             if (key != null) { key.free(); }
+            if (store != null) { store.free(); }
         }
     }
 
-    //FLEncodable
-    @Override
-    public void encodeTo(FLEncoder encoder) {
-        final Object info = encoder.getExtraInfo();
-        if (info != null) {
-            installInDatabase(((MutableDocument) info).getDatabase());
-        }
-
-        final Map<String, Object> dict = jsonRepresentation();
-        encoder.beginDict(dict.size());
-        for (Map.Entry<String, Object> entry : dict.entrySet()) {
-            encoder.writeKey(entry.getKey());
-            encoder.writeValue(entry.getValue());
-        }
-        encoder.endDict();
-    }
-
-    //---------------------------------------------
-    // Private (in class only)
-    //---------------------------------------------
-    private C4BlobStore getBlobStore() {
-        if (database == null) {
-            Log.w(DOMAIN, "database instance is null.");
-            return null;
-        }
+    @NonNull
+    private byte[] getBytesFromStream(@NonNull InputStream in) {
+        final ByteArrayOutputStream out = new ByteArrayOutputStream();
+        byte[] buff = new byte[MAX_CACHED_CONTENT_LENGTH];
         try {
-            return database.getBlobStore();
+            int n;
+            while ((n = in.read(buff)) > 0) { out.write(buff, 0, n); }
```

Comment:
From Java doc:
```
the total number of bytes read into the buffer, or -1 if there is no more data because the end of the stream has been reached.
```

Is it safer to check for -1 or >= 0?

Reply:
As above.  Fixing now.